overlord/snapstate/flags.go

```diff
@@ -50,3 +54,32 @@ func (f Flags) ForSnapSetup() Flags {
 	f.IgnoreValidation = false
 	return f
 }
+
+// EffectiveConfinement return the effective confinement for a given declared
+// confinement and flags like jailmode or devmode.
+func (f Flags) EffectiveConfinement(confinementType snap.ConfinementType) snap.ConfinementType {
+	switch confinementType {
+	case snap.ClassicConfinement:
+		// unchanged, flags don't impact snaps with classic confinement
```

Comment:
Why not?  If I say --jailmode on a classic snap, I'd expect it to be jailed. If I say --devmode, I'd expect it to be put in devmode.. _development_ mode.. might not affect confinement directly, but might affect other decisions.